db/db_impl/db_impl_files.cc

```diff
@@ -963,28 +942,24 @@ Status DBImpl::SetupDBId(const WriteOptions& write_options, bool read_only,
   return s;
 }
 
-Status DBImpl::DeleteUnreferencedSstFiles(RecoveryContext* recovery_ctx) {
-  mutex_.AssertHeld();
-  std::vector<std::string> paths;
-  paths.push_back(NormalizePath(dbname_ + std::string(1, kFilePathSeparator)));
+void DBImpl::CollectAllDBPaths() {
```

Comment:
With this functionality factored into a function of a giant class, I would like to see enforcement of preconditions. Specifically, either check that it hasn't been called before, or allow repeated calls with no ill effect.

Reply:
I added a set empty before populating check.